lib/classes/PluginManager.js

```diff
@@ -215,6 +220,29 @@ class PluginManager {
     );
   }
 
+  getLocalPluginsPathPatterns() {
+    return this.localPluginsPaths
+      .map(({ resolvedPath, inputPath }) => {
+        const absoluteInputPath = path.resolve(this.serverless.serviceDir, inputPath);
+        if (
+          absoluteInputPath === resolvedPath ||
+          absoluteInputPath + path.extname(resolvedPath) === resolvedPath
+        ) {
+          return path.relative(this.serverless.serviceDir, resolvedPath);
+        }
+
+        if (resolvedPath.startsWith(absoluteInputPath + path.sep)) {
+          return path.relative(
+            this.serverless.serviceDir,
+            path.join(path.dirname(resolvedPath), '/**')
+          );
+        }
+
+        return null;
+      })
+      .filter((v) => v !== null);
```

Comment:
We can also simplify it to `.filter(Boolean)`

Reply:
I was going to do that, but I felt you'd favour the explicitness. No harm leaving it this way, right?

Reply:
Yes, it's fine to leave it